core/src/components/checkbox/test/bottom-content/checkbox.e2e.ts

```diff
@@ -0,0 +1,205 @@
+    });
+    test('input should have aria-invalid attribute when checkbox is invalid', async ({ page }) => {
+      await page.setContent(
+        `<ion-checkbox class="ion-invalid ion-touched" helper-text="Helper text" error-text="Error text">Label</ion-checkbox>`,
+        config
+      );
+
+      const input = page.locator('ion-checkbox input[type=checkbox]');
+
+      await expect(input).toHaveAttribute('aria-invalid');
+    });
+    test('input should not have aria-invalid attribute when checkbox is valid', async ({ page }) => {
+      await page.setContent(
+        `<ion-checkbox helper-text="Helper text" error-text="Error text">Label</ion-checkbox>`,
+        config
+      );
+
+      const input = page.locator('ion-checkbox input[type=checkbox]');
+
+      await expect(input).not.toHaveAttribute('aria-invalid');
+    });
+    test('input should not have aria-describedby attribute when no hint or error text is present', async ({ page }) => {
+      await page.setContent(`<ion-checkbox>Label</ion-checkbox>`, config);
+
+      const input = page.locator('ion-checkbox input[type=checkbox]');
+
+      await expect(input).not.toHaveAttribute('aria-describedby');
+    });
+  });
+});
+
+/**
+ * Rendering is different across modes
+ */
+configs({ modes: ['ios', 'md'], directions: ['ltr'] }).forEach(({ title, screenshot, config }) => {
+  test.describe(title('checkbox: helper text rendering'), () => {
+    // Check the default label placement, end, and stacked
+    [undefined, 'end', 'stacked'].forEach((labelPlacement) => {
+      test(`${
+        labelPlacement ? `${labelPlacement} label - ` : ''
+      }should not have visual regressions when rendering helper text`, async ({ page }) => {
+        await page.setContent(
+          `<ion-checkbox ${
+            labelPlacement ? `label-placement="${labelPlacement}"` : ''
+          } helper-text="Helper text">Label</ion-checkbox>`,
+          config
+        );
+
+        const bottomEl = page.locator('ion-checkbox');
+        await expect(bottomEl).toHaveScreenshot(
+          screenshot(`checkbox-helper-text${labelPlacement ? `-${labelPlacement}` : ''}`)
+        );
+      });
+
+      test(`${
+        labelPlacement ? `${labelPlacement} label - ` : ''
+      }should not have visual regressions when rendering helper text with wrapping text`, async ({ page }) => {
+        await page.setContent(
+          `<ion-checkbox ${
+            labelPlacement ? `label-placement="${labelPlacement}"` : ''
+          } helper-text="Helper text helper text helper text helper text helper text helper text helper text helper text helper text">Label</ion-checkbox>`,
+          config
+        );
+
+        const bottomEl = page.locator('ion-checkbox');
+        await expect(bottomEl).toHaveScreenshot(
+          screenshot(`checkbox-helper-text${labelPlacement ? `-${labelPlacement}` : ''}-wrapping`)
+        );
+      });
+    });
+  });
+
+  test.describe(title('checkbox: error text rendering'), () => {
```

Comment:
Why does error text not get a check for the end placement?

Reply:
The error text checks are mainly making sure it shows up and the colors look right. The layout for the error text is the same as the helper text, but in the case of stacked it needs to still be centered so I checked that too. Honestly we could probably remove stacked for it, but I don't feel strongly either way if you want me to check all of the label placements.

Reply:
I'm fine with this. Thanks!